src/nl/hannahsten/texifyidea/util/parser/LatexPsiUtil.kt

```diff
@@ -77,7 +77,7 @@ fun LatexEndCommand.beginCommand(): LatexBeginCommand? = previousSiblingOfType(L
 /**
  * Checks if the latex content objects is a display math environment.
  */
-fun LatexNoMathContent.isDisplayMath() = firstChildOfType(LatexDisplayMath::class) != null && firstChildOfType(LatexEnvironment::class) == null
+fun LatexNoMathContent.isDisplayMath() = children.firstOrNull() is LatexMathEnvironment
```

Comment:
Thanks for taking a look at this! This can still be improved because LatexMathEnvironment can also be inline math (see `latex.bnf`). I guess the correct way would be to check the child of the LatexMathEnvironment.
I don't think the inspection should trigger on inline math, but not sure.

Reply:
Okay, it works but im sure youll want to tweak it. tests also added

Reply:
I think this should work, thanks!